### DISCUSSION TITLE
best approach to handling high volume event tracking?

### DISCUSSION THREAD

**Discussion by joeyscarim** [umami-software/umami#1310] [Q&A]
I'm currently using Umami with next.js + google cloud sql for general analytics, and redis (upstash + iosredis) for event tracking.

We run live virtual events, and track each session watched, so we can get up to ~5000 events to track all at once sometimes.

I'd like to move this event tracking into umami also, but I'm having trouble researching where/what my bottlenecks would be doing this

Any advice appreciated!

**Comment by prasannamestha** [comment]
I recently started using umami and I feel you. I use it for my personal blog that gets about 50-100 visitors a day, even for such a low traffic site I felt umami to be a bit slow in displaying charts. I believe it could be because of reading data from the database.

I believe if we find a way to hook redis we should be able to drastically improve the performance (and even handle higher volumes)

**Reply by prasannamestha** [reply]
I just found out that adding `REDIS_URL` to env integrates redis. Oops..
